packages/react/src/categories/hooks/useCategory.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  fetchCategory as fetchCategoryAction,
+  getCategory,
+  getCategoryError,
+  isCategoryFetched,
+  isCategoryLoading,
+  StoreState,
+} from '@farfetch/blackout-redux';
+import { useAction } from '../../helpers';
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import type { UseCategoryOptions } from './types/useCategory';
+
+const useCategory = (id: number, options: UseCategoryOptions = {}) => {
+  const { enableAutoFetch = true, fetchConfig } = options;
+  const error = useSelector((state: StoreState) => getCategoryError(state, id));
+  const isLoading = useSelector((state: StoreState) =>
+    isCategoryLoading(state, id),
+  );
+  const isFetched = useSelector((state: StoreState) =>
+    isCategoryFetched(state, id),
+  );
+  const category = useSelector((state: StoreState) => getCategory(state, id));
+  const fetch = useAction(fetchCategoryAction);
+
+  useEffect(() => {
+    if (!isLoading && !error && !isFetched && enableAutoFetch)
+      fetch(id, fetchConfig);
+  }, [
+    category,
+    enableAutoFetch,
+    error,
+    fetch,
+    fetchConfig,
+    id,
+    isFetched,
+    isLoading,
+  ]);
+
+  return {
+    isLoading,
+    error,
+    isFetched,
+    data: category,
+    actions: {
+      fetch,
```

Comment:
IMO we should discuss this approach. I believe we have both being applied on our react custom hooks.

Reply:
Just to add to this, I think Pedro's comment is correct and all hooks that do not follow that approach should be changed to conform to it. The reason is that the hook provides its result from the passed in parameter instead, so if the `fetch` action allows to request a different `id`, the developer will not have access to its result until he changed the `id` parameter of the hook, but if he does that, the `useEffect` will be triggered which will do the fetch anyway (if he has `enableAutoFetch` true of course).

I only think that it should be named `fetch` instead of `refetch` because if the `enableAutoFetch` parameter is false, `refetch` I think does not make much sense.

Another point is that the `fetch` action should be provided by the `useCategories` hook and this hook should depend on that. The `fetch` provided by the `useCategory` hook would only set the `id` parameter to the `id` parameter passed to the hook.

What do you think?

Reply:
Done! The `fetch` action to request a different ` id`, was removed! 

Regarding your 2nd point: we have 2 different endpoints: one for the list of categories ('`/commerce/v1/categories`'), and another to fetch a category ('`/commerce/v1/categories/:id`'). So, I think that makes sense to have 2 different actions! =)